proto/substrait/algebra.proto

```diff
@@ -178,6 +178,18 @@ message JoinRel {
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 }
 
+// A time-series variant of the multi JOIN relational operator left-join-right(s), which joins on an ordered key using inexact matching
+message AsOfJoinRel {
+  JoinRel join = 1;
+  int64 tolerance = 2;
+  oneof on_type {
+    Expression.FieldReference on = 3;
+    // Reserve next tags for future on_type alternatives
+  }
```

Comment:
What would be an example of a different `on_type`?

Reply:
A natural example is an algebraic expression involving one or more fields and returning a value in an ordered domain. Another, perhaps contrived example, is multiple on-fields interpreted as a radix.